src/metatensor/models/cli/eval_model.py

```diff
@@ -154,5 +166,17 @@ def eval_model(
         _eval_targets(model, eval_dataset)
 
     # Predict structures
-    predictions = model(eval_structures, model.capabilities.outputs)
+    # TODO: batch this
+    # TODO: add forces/stresses/virials if requested
+    if not hasattr(options, "targets"):
```

Comment:
This conditions is a bit hard to understand. Why do we to check for targets and options to calculate the NLs (or not)? AFAIK, the options doesn't contain any information about the NHS, especially if we need to compute them or not... Please correct me, if I'm wrong.

Reply:
```

>  # Predict targets
>     if hasattr(options, "targets"):
>         eval_targets = read_targets(conf=options["targets"], dtype=dtype)
>         eval_dataset = Dataset(structure=eval_structures, energy=eval_targets["energy"])
>         _eval_targets(model, eval_dataset)

```

For instance, in the same file just a few lines above we're evaluating the targets, but I guess the NLs in the case of Alchemical Model will be also required

Reply:
At the moment this section is a big hack with a lot of TODOs. It will crash for large datasets because we don't batch and it won't do gradients (no forces/virials/stresses). I was going to open an issue as soon as this is merged.

Regarding the reason why we don't do NLs if "targets" is included:
if there are "targets", it means that we're comparing against the xyz file (most likely to get an RMSE) rather than calculating unknown properties. In the block above, we calculate those RMSEs and then we go on to save a file with the new properties. In that case, the systems will already have the NLs from the RMSE calculation above.

In my opinion, the whole structure of eval needs to change. The cleanest option would be to save the predictions to a file and then calculate the RMSEs from the two files (user-provided vs saved-by-us). In any case, it should work with any model (including the alchemical model) because the NLs are available and stored in the Systems from the RMSE calculation